example/src/main/java/co/infinum/example/MainActivity.kt

```diff
@@ -256,6 +334,37 @@ class MainActivity : AppCompatActivity() {
             start()
         }
     }
-}
 
-object SmartassException : Throwable()
+    private fun allPermissionsGranted() = REQUIRED_PERMISSIONS.all {
+        ContextCompat.checkSelfPermission(
+            baseContext, it
+        ) == PackageManager.PERMISSION_GRANTED
+    }
+
+    fun getOutputDirectory(): File {
+        val mediaDir = externalMediaDirs.firstOrNull()?.let {
+            File(it, resources.getString(R.string.app_name)).apply { mkdirs() }
+        }
+        return if (mediaDir != null && mediaDir.exists())
+            mediaDir else filesDir
+    }
+
+    private fun setUpPinchToZoom() {
```

Comment:
Is this something that was included in GoldenEye, but you had to include manually due to CameraX not supporting it by default?

Reply:
Yes, with GoldenEye you just need to set callback what you want to happened when zoom changes.
The good thing with camerax is that you have more control on zoom as far as I see https://proandroiddev.com/android-camerax-tap-to-focus-pinch-to-zoom-zoom-slider-eb88f3aa6fc6

Reply:
I definitely prefer GoldenEye here, but I suppose allowing for a bit more control has its merits (even though the purpose of the lib is simplicity rather than customizability).